src/software/ai/evaluation/calc_best_shot.h

```diff
@@ -10,6 +10,372 @@
 #include "software/world/team.h"
 #include "software/world/world.h"
 
+class AngleSegment
+{
+   public:
+    /**
+     * Constructs an AngleSegment represented by a top angle and bottom angle
+     *
+     * @param angle_top the most north angle
+     * @param angle_bottom the most south angle
+     */
+    AngleSegment(Angle angle_top, Angle angle_bottom)
+        : angle_top_(angle_top), angle_bottom_(angle_bottom)
+    {
+    }
+
+    /**
+     * Gets the top (most north) angle of the angle segment
+     *
+     * @return the top angle of the angle segment
+     */
+    Angle getAngleTop()
+    {
+        return angle_top_;
+    }
+
+    /**
+     * Sets the top (most north) angle of the angle segment
+     *
+     * @param angle_top the top angle of the angle segment
+     */
+    void setAngleTop(Angle angle_top)
+    {
+        this->angle_top_ = angle_top;
+    }
+
+    /**
+     * Gets the bottom (most south) angle of the angle segment
+     *
+     * @return the bottom angle of the angle segment
+     */
+    Angle getAngleBottom()
+    {
+        return angle_bottom_;
+    }
+
+    /**
+     * Sets the bottom (most south) angle of the angle segment
+     *
+     * @param angle_bottom the bottom angle of the angle segment
+     */
+    void setAngleBottom(Angle angle_bottom)
+    {
+        this->angle_bottom_ = angle_bottom;
+    }
+
+    /**
+     * Gets the abs delta between the two angles that describe this angle segment
+     *
+     * @return the abs delta between the top and bottom angles
+     */
+    double getDelta()
+    {
+        return (angle_bottom_ - angle_top_).abs().toDegrees();
+    }
+
+    /**
+     * Compares another AngleSegment's top angle
+     * Used to make sorting more performant
+     *
+     * @param other the other AngleSegment
+     * @return true if both AngleSegment's top angles are equal
+     */
+    bool operator==(const AngleSegment &other) const
+    {
+        return other.angle_top_ == angle_top_;
+    }
+
+    /**
+     * Compares another AngleSegment's top angle
+     * Used to make sorting more performant
+     *
+     * @param other the other AngleSegment
+     * @return true if this top angle is less than the other's top angle
+     */
+    bool operator<(const AngleSegment &other) const
+    {
+        return angle_top_ < other.angle_top_;
+    }
+
+    /**
+     * Compares another AngleSegment's top angle
+     * Used to make sorting more performant
+     *
+     * @param other the other AngleSegment
+     * @return true if this top angle is greater than the other's top angle
+     */
+    bool operator>(const AngleSegment &other) const
+    {
+        return angle_top_ > other.angle_top_;
+    }
+
+   private:
+    Angle angle_top_;
+    Angle angle_bottom_;
+};
+
+/**
+ *
+ */
+class AngleMap
```

Comment:
I'd like to see geom-like classes like AngleSegment, AngleMap, etc be put in the geom folder as their own classes  with unit tests

Reply:
separated AngleSegment. AngleMap requires a special extension of AngleSegment (ObstacleAngleSegment) so it's not easily separated into geom